ietf/doc/views_ballot.py

```diff
@@ -337,11 +344,24 @@ def send_ballot_comment(request, name, ballot_id):
 
     balloter = request.user.person
 
-    if 'ballot_edit_return_point' in request.session:
-        return_to_url = request.session['ballot_edit_return_point']
-    else:
-        return_to_url = urlreverse("ietf.doc.views_doc.document_ballot", kwargs=dict(name=doc.name, ballot_id=ballot_id))
+    return_to_url = request.GET.get('ballot_edit_return_point')
 
+    if return_to_url is not None:
+        # we need to ensure return_to_url isn't used for phishing attacks.
+        # return_to_url is used in HttpResponseRedirect() which could redirect to Datatracker or offsite.
+        # Eg http://datatracker.ietf.org/?ballot_edit_return_point=https://example.com/phishing-attack
+        # offsite links could be phishing attempts so let's reject them all, and require valid Datatracker
+        # routes
+        try:
+            urlresolve(return_to_url)
```

Comment:
I wonder if we should instead (or maybe in addition) check that the URL is not absolute? I think that'd prevent the phishing issue.

I wonder if it'd be worth putting an even tighter check here and only redirect to the view that we want. E.g., someone could make a nuisance link that'd cause logout (at least, until we disable log out by GET). I worry that there might be more harmful games possible with this.

Reply:
> check that the URL is not absolute?

yeah I thought about checking whether it was a relative vs absolute URL through checking if it started with a `/` but then there's that protocol-relative URL syntax of `//example.com/phish` ([now considered an antipattern](https://stackoverflow.com/questions/28446314/why-use-protocol-relative-urls-at-all) so I'd have to use a real URL parser to be safe, and then if I were checking for valid routes by string that's a subset of relative URLs so it didn't feel necessary.

> only redirect to the view that we want

there are currently 5 views that assign `request.session['ballot_edit_return_point']` so I'm guessing there are 5 ways into the ballot editing and 5 ways we should redirect out. We could have an allow list of those 5 route patterns.

Reply:
^^ working on this now

Reply:
Done. Thoughts?